code/datums/martial/boxing.dm

```diff
@@ -31,23 +40,40 @@
 
 ///Unlike most instances of this proc, this is actually called in _proc/tussle()
 ///Returns a multiplier on our skill damage bonus.
-/datum/martial_art/boxing/proc/check_streak(mob/living/attacker, mob/living/defender)
-	var/combo_multiplier = 1
+/datum/martial_art/boxing/proc/check_streak(mob/living/attacker, mob/living/defender, obj/item/bodypart/arm/active_arm)
+	if(check_behind(attacker, defender) && !honorable_boxer)
+		reset_streak()
+		return CRAVEN_BLOW
+
+	if(is_detective_job(attacker.mind?.assigned_role)  && defender.is_blind()) //In short: discombobulate
+		reset_streak()
+		return DISCOMBOBULATE
 
-	if(findtext(streak, LEFT_LEFT_COMBO) || findtext(streak, RIGHT_RIGHT_COMBO))
+	if(findtext(streak, LEFT_LEFT_COMBO) && active_arm.body_zone == BODY_ZONE_R_ARM || findtext(streak, RIGHT_RIGHT_COMBO) && active_arm.body_zone == BODY_ZONE_L_ARM)
 		reset_streak()
-		if(COOLDOWN_FINISHED(src, warning_cooldown))
-			COOLDOWN_START(src, warning_cooldown, 2 SECONDS)
-			attacker.balloon_alert(attacker, "weak combo, alternate your hits!")
-		return combo_multiplier * 0.5
+		if(attacker.is_blind())
+			return BLIND_JAB
+		else
+			return LIGHT_JAB
+
+	else if(findtext(streak, LEFT_LEFT_COMBO) && active_arm.body_zone == BODY_ZONE_L_ARM || findtext(streak, RIGHT_RIGHT_COMBO) && active_arm.body_zone == BODY_ZONE_R_ARM)
+		reset_streak()
+		return STRAIGHT_PUNCH
 
 	if(findtext(streak, LEFT_RIGHT_COMBO) || findtext(streak, RIGHT_LEFT_COMBO))
 		reset_streak()
-		// If we have an extra effect from the combo, perform it here. By default, we have no extra effect.
-		perform_extra_effect(attacker, defender)
-		return combo_multiplier * 1.5
+		if(active_arm.body_zone == BODY_ZONE_L_ARM)
+			if(findtext(streak, RIGHT_LEFT_COMBO))
+				return LEFT_HOOK
 
-	return combo_multiplier
+		else if(active_arm.body_zone == BODY_ZONE_R_ARM)
+			if(findtext(streak, LEFT_RIGHT_COMBO))
+				return RIGHT_HOOK
+		else
+			return UPPERCUT
+
+	perform_extra_effect(attacker, defender)
```

Comment:
Should this really be in `check_streak` and not `tussle`? it wouldn't be immediately obvious to me that calling check streak can result in side effects